code/API_definitions/Optimal-Edge-Discovery.yaml

```diff
@@ -0,0 +1,741 @@
+tags:
+  - name: Regions
+    description: |
+      Get all Region IDs and Names for edge cloud zones
+  - name: Discovery
+    description: |
+      Discover the regions and zones in the edge cloud
+      application, find optimal edge cloud zones for
+      your deployed applications, and optimal
+      Application Endpoints for your clients to connect to.
+############################################################################
+#                                     Paths                                #
+############################################################################
+paths:
+  /regions:
+    get:
+      summary: Fetch all Regions
+      operationId: get-regions
+      parameters:
+        - name: x-correlator
+          in: header
+          required: false
+          description: |
+            When the API Consumer includes the "x-correlator" header in the
+            request, the API provider must include it in the response with
+            the same value that was used in the request. Otherwise, it is
+            optional to include the "x-correlator" header in the response with
+              any valid value. Recommendation is to use UUID for values.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Returns Edge Cloud Regions
+          content:
+            application/json:
+              schema:
+                title: GetEdgeCloudRegionsResponse
+                type: array
+                items:
+                  $ref: "#/components/schemas/EdgeCloudRegions"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/Generic404"
+        "406":
+          $ref: "#/components/responses/Generic406"
+        "422":
+          $ref: "#/components/responses/Generic422"
+        "429":
+          $ref: "#/components/responses/Generic429"
+      tags:
+        - Regions
+      description: |
+        This is a helper API to get all the regions. The response identifies the
+        regions where edge cloud zones are available.
+  /edge-cloud-zones:
+    post:
+      operationId: get-edgeCloudZone
+      parameters:
+        - name: x-correlator
+          in: header
+          schema:
+            type: string
+            description: |
+              A unique identifier for the request, which can be used for
+              troubleshooting purposes. When the API Consumer includes the
+              "x-correlator" header in the request, the API provider must
+              include it in the response with the same value that was used in
+              the request. Otherwise, it is optional to include the
+              "x-correlator" header in the response with any valid value.
+              Recommendation is to use UUID for values.
+
```

Comment:
```suggestion
        - name: x-correlator
          in: header
          required: false
          description: |
            When the API Consumer includes the "x-correlator" header in the
            request, the API provider must include it in the response with
            the same value that was used in the request. Otherwise, it is
            optional to include the "x-correlator" header in the response with
             any valid value. Recommendation is to use UUID for values.
          schema:
            type: string
            pattern: ^[a-zA-Z0-9-]{0,55}$
```